tests/drivers/clock_control/clock_control_api/src/test_clock_control.c

```diff
@@ -14,97 +14,117 @@ LOG_MODULE_REGISTER(test);
 
 struct device_data {
 	const char *name;
-	u32_t startup_us;
+	u32_t subsys_cnt;
+	const u32_t *subsys_startup_us;
 };
 
 static const struct device_data devices[] = {
 #ifdef DT_INST_0_NORDIC_NRF_CLOCK_LABEL
 	{
-		.name = DT_INST_0_NORDIC_NRF_CLOCK_LABEL  "_16M",
-		.startup_us = IS_ENABLED(CONFIG_SOC_SERIES_NRF91X) ? 3000 : 400
-	},
-
-	{
-		.name = DT_INST_0_NORDIC_NRF_CLOCK_LABEL  "_32K",
-		.startup_us = (CLOCK_CONTROL_NRF_K32SRC == NRF_CLOCK_LFCLK_RC) ?
-				1000 : 300000
+		.name = DT_INST_0_NORDIC_NRF_CLOCK_LABEL,
+		.subsys_cnt = NRF_CLOCK_TYPE_COUNT,
+		.subsys_startup_us = (const u32_t[]){
+			/* hfclk */
+			IS_ENABLED(CONFIG_SOC_SERIES_NRF91X) ? 3000 : 400,
```

Comment:
Same here to comments in the nrf clock driver itself. Could this use indexes in the initializer to not need the comments?

Reply:
done